src/server/utils/bucket_diff.js

```diff
@@ -151,33 +171,42 @@ class BucketDiff {
      * @param {import("aws-sdk/lib/request").PromiseResult<AWS.S3.ListObjectVersionsOutput, AWS.AWSError> | 
      *         import("aws-sdk/lib/request").PromiseResult<AWS.S3.ListObjectsV2Output, AWS.AWSError>} list
      * 
-     * _object_grouped_by_key_and_omitted will return the objects grouped by key.
+     * _object_grouped_by_key_and_omitted will return the objects and delete markers grouped by key.
      * When we have versioning enabled, if there is more than one key, it omits 
-     * the last key from the object, in order to avoid processing incomplete list of object + version
+     * the last key from both, in order to avoid processing incomplete list of object + version
      * 
-     * the results of this will be {{ [key: string]: Array<object> }} where the Array consist of metadata objects.
+     * the results of this will be two objects of type {{ [key: string]: Array<object> }} where the Array consist of metadata objects.
      *       for example: {
      *            "1": [
      *                { ETag: 'etag1.1', Size: 24599, Key: '1', VersionId: 'v1.1', IsLatest: true, },
      *                { ETag: 'etag1.2', Size: 89317, Key: '1', VersionId: 'v1.2', IsLatest: false, }
      *            ]
      *        }
-     * @returns {nb.BucketDiffKeysDiff}
+     * @returns {Object}
      */
     _object_grouped_by_key_and_omitted(list) {
         if (!list) return {};
         dbg.log1('_object_grouped_by_key_and_omitted list:', list);
         const field = this.version ? "Versions" : "Contents";
-        let grouped_by_key = _.groupBy(list[field], "Key");
+        const delete_marker = 'DeleteMarkers';
+
+        // group objects and delete markers with the Key
+        let grouped_objects = _.groupBy(list[field], "Key");
+        let grouped_delete_markers = list[delete_marker] && _.groupBy(list[delete_marker], "Key");
         // We should not omit if this is a list object and not list versions
         // and the use of continuation token later on the road will lead us to skip the last key if omitted.
         if (list.IsTruncated && this.version) {
-            const last_key_pos = list[field].length - 1;
-            if (Object.keys(grouped_by_key).length > 1) {
-                grouped_by_key = _.omit(grouped_by_key, list[field][last_key_pos].Key);
+            let last_key_pos = list[field].length - 1;
+            if (Object.keys(grouped_objects).length > 1) {
+                grouped_objects = _.omit(grouped_objects, list[field][last_key_pos].Key);
+            }
+
+            last_key_pos = list[delete_marker].length - 1;
+            if (Object.keys(grouped_delete_markers).length > 1) {
+                grouped_delete_markers = _.omit(grouped_delete_markers, list[delete_marker][last_key_pos].Key);
```

Comment:
I think we should only trancate the last delete marker key if it is equal to the last key in the version field.